src/map/ai/helpers/pathfind.cpp

```diff
@@ -580,7 +580,16 @@ float CPathFind::GetRealSpeed()
         }
         else if (m_POwner->animation == ANIMATION_ATTACK)
         {
-            realSpeed = realSpeed + settings::get<int8>("map.MOB_SPEED_MOD");
+            auto speedMod = settings::get<int8>("map.MOB_SPEED_MOD");
+            if (speedMod + realSpeed > 0)
+            {
+                realSpeed = realSpeed + speedMod;
+            }
+            else
+            {
+                // don't let speed underflow due to extreme map setting or low base mob speed
+                realSpeed = 1;
+            }
```

Comment:
speed can be exactly zero in some legitimate cases. your conditionals handled > than zero and then in all other cases sets to exactly one.

Reply:
I want to say speed zero would never get to this function, but I'll do some more testing

Reply:
After discussing with Teo, I decided to convert this map setting to a %. The logic is so much simpler and it honestly makes more sense since the global `SPEED_MOD` is raw for all entities. 

Cool thing I learned: map settings file reloads go into effect real-time. You can set `MOB_SPEED_MOD` to `-100`, pull a mob, then save different values to watch the mob change run speed